### PR TITLE
fixed issue #1078 Add Report on learner name,Model parameter , Data on Stochastic Gradient Regression.

### PR DESCRIPTION
**Pull request by shark-S**
branch for issue 1078 . issue is fixed now

### CODE REVIEW

**Comment by shark-S** [comment]
@ajdapretnar  and @kernc  .Now report is working  learner name , Model parameter and data

**Comment by janezd** [comment]
Please use descriptive titles. Seeing "fixed issue 1078" in the list of pull requests doesn't tell what the PR is about. You can write "fixed issue #1078" in the description, though, so that github adds the links from issue to PR and back.

Reports must look nice and readable. `squared_epsilon_insensitive` and `invscaling` are for coders, not users. Please see what other reports look like. It's HTML, not json. :)

By the way, this particular issue is already almost done in #1107.

**Comment by shark-S** [comment]
okk @janezd  next time i will take care of it . i have make a previous  PR  #1098 in this but that is wrong now it is corrected so i made a PR .

**Comment by janezd** [comment]
You can still edit the title if you plan to continue this exercise. If you don't, you can close the pull request. (Hm, or does one need to have write permissions for this?)

**Comment by shark-S** [comment]
@janezd  while i am renaming this branch it can be "fixedissue#1078" but can't "fixed issue #1078" suggest me.i can't understand by write permission can you explain a little bit more?

**Comment by janezd** [comment]
Don't rename the branch, rename the pull request. Click Edit at the top of this page.

**Comment by shark-S** [comment]
i have changed it but here it not refer to its link . Should i have to give there a link?

**Comment by kernc** [comment]
https://guides.github.com/features/issues/#notifications

**Comment by kernc** [comment]
https://guides.github.com/features/issues/#notifications

Pull request titles should be likened to that of [commit messages](https://github.com/biolab/orange3/blob/master/CONTRIBUTING.md#commit-messages).

**Comment by shark-S** [comment]
@janezd and @kernc   i have improved  Report please check and suggest i need to change anything.

**Comment by kernc** [comment]
There's [250 lines of changes](https://github.com/biolab/orange3/pull/1111/files), I don't think that's correct. Please squash the commits and rebase on `upstream/master`.

**Comment by shark-S** [comment]
@kernc  thanks for help.Report Improved :)

File: Orange/widgets/regression/owsgdregression.py [hunk 1]
```diff
@@ -130,6 +129,9 @@ def __init__(self):
         self._on_func_changed()
 
         self.apply()
+        
+
+
```
Comment (kernc) [inline Orange/widgets/regression/owsgdregression.py]: Prefer not to introduce extra vertical whitespace when it's not needed. This is also not PEP8-compliant, try by running:
```bash
$ sudo apt-get install pep8
$ pep8 Orange/widgets/regression/owsgdregression.py
```

File: Orange/widgets/regression/owsgdregression.py [hunk 2]
```diff
@@ -196,7 +198,22 @@ def _on_lrate_changed(self):
         mask = enabled[self.learning_rate]
         for spin, enabled in zip(self._lrate_params, mask):
             spin.setEnabled(enabled)
-
+    def send_report(self):
+        self.report_items((("Name", self.learner_name),))
+        items=OrderedDict()
```
Comment (kernc) [inline Orange/widgets/regression/owsgdregression.py]: Missing whitespace around equals sign.

File: Orange/widgets/regression/owsgdregression.py [hunk 3]
```diff
@@ -196,7 +198,22 @@ def _on_lrate_changed(self):
...
+
+        items['Loss Function'] = ["Squared loss", "Huber", "Epsilon Insensitive", "Squared Epsilon Insensitive"][self.loss_function]
+        items['Penalty'] = ["Absolute norm (L1)", "Euclidean norm (L2)", "Elastic Net (both)"][self.penalty_type]
+        items['Learning Rate'] = ["Inverse scaling", "Constant"][self.learning_rate]
```
Comment (kernc) [inline Orange/widgets/regression/owsgdregression.py]: Please ensure lines break at or before 80th column.

File: Orange/widgets/regression/owsgdregression.py [hunk 4]
```diff
@@ -196,7 +198,22 @@ def _on_lrate_changed(self):
...
+        items['No of iterations']=self.n_iter
+        self.report_items("Model paremeter",items)
+        if self.data:
+            self.report_data("Data", self.data)
+        
+         
+                           
+        
+
```
Comment (kernc) [inline Orange/widgets/regression/owsgdregression.py]: Please strip this extra whitespace too.

File: Orange/widgets/regression/owsgdregression.py [hunk 5]
```diff
@@ -196,7 +198,22 @@ def _on_lrate_changed(self):
...
             spin.setEnabled(enabled)
-
+    def send_report(self):
+        self.report_items((("Name", self.learner_name),))
+        items=OrderedDict()
+
+        items['Loss Function'] = ["Squared loss", "Huber", "Epsilon Insensitive", "Squared Epsilon Insensitive"][self.loss_function]
+        items['Penalty'] = ["Absolute norm (L1)", "Euclidean norm (L2)", "Elastic Net (both)"][self.penalty_type]
+        items['Learning Rate'] = ["Inverse scaling", "Constant"][self.learning_rate]
+        items['No of iterations']=self.n_iter
```
Comment (kernc) [inline Orange/widgets/regression/owsgdregression.py]: Spaces around `=`.

File: Orange/widgets/regression/owsgdregression.py [hunk 6]
```diff
@@ -42,7 +40,7 @@ class OWSGDRegression(OWProvidesLearner, widget.OWWidget):
     learning_rate = settings.Setting(InvScaling)
 
     want_main_area = False
-    resizing_enabled = False
+    resizing_enabled = True
```
Comment (kernc) [inline Orange/widgets/regression/owsgdregression.py]: While I understand your motivation here, this is not the right way to do it. Please revert this from this PR. If you feel like fixing this widget being too high, consider doing it by relayouting its contents into two columns.

Comment (shark-S) [inline Orange/widgets/regression/owsgdregression.py]: @kernc  thanks for guide it is good idea. I have made some silly mistake like whitespace,spaces.I will improve them .

Comment (kernc) [inline Orange/widgets/regression/owsgdregression.py]: No problem, that's what code review is for. But if you do opt to relayout, please do it in a separate PR. :smiley:

Comment (shark-S) [inline Orange/widgets/regression/owsgdregression.py]: okk so should i have to close this pull request?

Comment (kernc) [inline Orange/widgets/regression/owsgdregression.py]: Not for the reasons above, no. This PR is still about a valid issue, namely reporting in SGD widget. However, now that I look at it, it's a duplicate of https://github.com/biolab/orange3/pull/1107.

Comment (shark-S) [inline Orange/widgets/regression/owsgdregression.py]: but how i have try to implement it as another report like [owsvmclassification.py](https://github.com/biolab/orange3/blob/master/Orange/widgets/classify/owsvmclassification.py)
.

Comment (kernc) [inline Orange/widgets/regression/owsgdregression.py]: Yes, the reporting function here is fine! (Save for a few minor issues as noted in the other comments.)

**Comment by janezd** [comment]
This is a good start, yet the PR #1107 is already almost there, so I'm closing this PR.

**Comment by shark-S** [comment]
okk @janezd , Actually i am working on nomograms as you said in [mail](https://groups.google.com/d/msgid/orange-gsoc/CAFh6oAycA1Lu9v0y%3D%2Bv6Yh8-f_Qd79G%2B0Bp%2BVkx-ALVhJeentw%40mail.gmail.com).  can i send a PR with improve widget ( with two columns) ? because it help me to familiar with it .

**Comment by janezd** [comment]
How is this related to this issue?

Nomogram is a complex widget, which also requires other components (in particular, naive Bayes, probably with LOESS) to be implemented first. This is not a simple pre-GSoC test issue, but a serious GSoC task.

**Comment by shark-S** [comment]
sorry for confusion .Nomograms is not related to this issue.
